activerecord/test/cases/associations/has_many_through_associations_test.rb

```diff
@@ -1359,6 +1359,18 @@ def test_insert_records_via_has_many_through_association_with_scope
     assert_equal [member], club.favorites
   end
 
+  def test_insert_records_via_has_many_through_association_with_scope_and_association_name_different_from_the_joining_table_name
```

Comment:
The test would fail on main with:
```diff
--- expected
+++ actual
@@ -1 +1 @@
-[#<Member id: 4, name: nil, member_type_id: nil, admittable_type: nil, admittable_id: nil>]
+#<ActiveRecord::Associations::CollectionProxy []>
```
without introduced fix